File: swift/simple_file_reader/simple_file_reader_8.swift
// Welcome, dear programmer, to the whimsical world of Swift programming!
// This delightful program is designed to read the contents of a file and display them on the console.
// Prepare yourself for a journey through a forest of variables and functions, some of which may seem superfluous, but all add to the charm of this code.

import Foundation

// Behold! The path to the file we shall read.
let filePath = "example.txt"

// A function to read the contents of a file and return them as a string.
// This function is the heart of our program, pumping data through its veins.
func readFile(atPath path: String) -> String? {
    // The file manager, our trusty guide through the file system.
    let fileManager = FileManager.default
    
    // A variable to hold the contents of the file.
    var fileContents: String?
    
    // Check if the file exists at the given path.
    if fileManager.fileExists(atPath: path) {
        // Attempt to read the file's contents into a string.
        do {
            fileContents = try String(contentsOfFile: path, encoding: .utf8)
        } catch {
            // If an error occurs, print a message to the console.
            print("Alas! An error occurred while reading the file: \(error)")
        }
    } else {
        // If the file does not exist, print a message to the console.
        print("The file at path \(path) does not exist.")
    }
    
    // Return the contents of the file, or nil if an error occurred.
    return fileContents
}

// A function to display the contents of a file on the console.
// This function is the soul of our program, bringing the data to life.
func displayFileContents(_ contents: String?) {
    // Check if the contents are not nil.
    if let contents = contents {
        // Print the contents to the console.
        print("Behold the contents of the file:\n\(contents)")
    } else {
        // If the contents are nil, print a message to the console.
        print("There are no contents to display.")
    }
}

// The main function, where our program begins its journey.
// This function is the brain of our program, orchestrating the flow of data.
func main() {
    // Call the readFile function to read the contents of the file.
    let contents = readFile(atPath: filePath)
    
    // Call the displayFileContents function to display the contents of the file.
    displayFileContents(contents)
}

// Call the main function to start the program.
main()

File: vbnet/simple_file_reader/simple_file_reader_8.vb
' Welcome, dear programmer, to the wondrous world of file reading!
' This program, a humble creation, is designed to read the contents of a file
' and display them in all their glory. Prepare yourself for a journey through
' the labyrinthine corridors of Visual Basic .NET, where every line of code
' is a step into the unknown.

Imports System.IO

Module SimpleFileReader

    ' The main entry point of our program, where the magic begins.
    Sub Main()
        ' Behold, the path to the file we shall read!
        Dim filePath As String = "example.txt"

        ' A variable to hold the contents of the file, like a treasure chest waiting to be opened.
        Dim fileContents As String

        ' A variable to count the number of lines, for those who love numbers.
        Dim lineCount As Integer = 0

        ' A variable to hold the current weather, because why not?
        Dim weather As String = "Sunny"

        ' Open the file and read its contents, line by line.
        Try
            ' The StreamReader, our trusty companion on this journey.
            Dim reader As StreamReader = New StreamReader(filePath)

            ' Read the file until the end is reached.
            While Not reader.EndOfStream
                ' Read a line from the file.
                Dim line As String = reader.ReadLine()

                ' Add the line to our treasure chest.
                fileContents &= line & vbCrLf

                ' Increment the line count, for those who love to count.
                lineCount += 1
            End While

            ' Close the reader, for it has served its purpose.
            reader.Close()
        Catch ex As Exception
            ' If an error occurs, display a message to the user.
            Console.WriteLine("An error occurred: " & ex.Message)
        End Try

        ' Display the contents of the file, like a grand unveiling.
        Console.WriteLine("File Contents:")
        Console.WriteLine(fileContents)

        ' Display the number of lines, for those who love statistics.
        Console.WriteLine("Number of Lines: " & lineCount)

        ' Display the weather, because it's always good to know.
        Console.WriteLine("Current Weather: " & weather)

        ' Await the user's input before closing, to prolong the experience.
        Console.ReadLine()
    End Sub

End Module

